common/types/ballot.go

```diff
@@ -230,21 +230,19 @@ func (b *Ballot) Initialize() error {
 		return fmt.Errorf("ballot already initialized")
 	}
 	if b.Signature == EmptyEdSignature {
-		return fmt.Errorf("cannot calculate Ballot ID: signature is nil")
+		return fmt.Errorf("Ballot must be signed before initialization")
 	}
 
 	if b.MsgHash != BytesToHash(b.HashInnerBytes()) {
 		return fmt.Errorf("bad message hash")
 	}
 
 	h := hash.New()
-	_, err := codec.EncodeTo(h, &b.InnerBallot)
-	if err != nil {
-		return fmt.Errorf("failed to encode inner ballot for hashing")
+	if _, err := h.Write(b.MsgHash[:]); err != nil {
+		return fmt.Errorf("failed to write to hash")
 	}
-	_, err = scale.EncodeByteSlice(scale.NewEncoder(h), b.Signature[:])
-	if err != nil {
-		return fmt.Errorf("failed to encode byte slice")
+	if _, err := scale.EncodeByteSlice(scale.NewEncoder(h), b.Signature[:]); err != nil {
+		return fmt.Errorf("failed to encode signature")
```

Comment:
don't change anything for ballot in this PR.

Reply:
I can move this to its own separate PR. I just noticed that a ballot is hashed twice when calling `Initialize`

Reply:
PR: https://github.com/spacemeshos/go-spacemesh/pull/new/ballot-initialize-double-hash